proxy/src/main/java/com/velocitypowered/proxy/command/builtin/ShutdownCommand.java

```
@@ -52,11 +54,22 @@ public static BrigadierCommand command(final VelocityServer server) {
                 StringArgumentType.greedyString())
             .executes(context -> {
               String reason = context.getArgument("reason", String.class);
-              server.shutdown(true, MiniMessage.miniMessage().deserialize(
-                  MiniMessage.miniMessage().serialize(
-                      LegacyComponentSerializer.legacy('&').deserialize(reason)
-                  )
-              ));
+              Component reasonComponent = null;
+
+              if (reason.startsWith("{") || reason.startsWith("[") || reason.startsWith("\"")) {
```

maybe we should do it like this?

```java
        char firstChar = reason.charAt(0);
        char lastChar = reason.charAt(reason.length() - 1);

        if ((firstChar == '{' && lastChar == '}')
                || (firstChar == '[' && lastChar == ']')
                || (firstChar == '"' && lastChar == '"')) {
                try {
                    reasonComponent = GsonComponentSerializer.gson()
                            .deserializeOrNull(reason);
                } catch (JsonSyntaxException e) {
                    log.warn("Failed to parse reason as JSON object: {}", reason, e);
                }
        }
```

---

generally not fond of printing a warning for that kinda thing, and generally not sure what exactly this would solve vs the current approach

---

this will allow server administrators to know the exact error when formatting is incorrect

---

> generally not fond of printing a warning for that kinda thing, and generally not sure what exactly this would solve vs the current approach


this will allow server administrators to know the exact error when formatting is incorrect

---

The command accepts json or MiniMessage, the intent is that it will silently fall through to parsing the alt format rather than blowing up

---

> The command accepts json or MiniMessage, the intent is that it will silently fall through to parsing the alt format rather than blowing up

How then to quickly determine where the error occurred? Maybe in that case it is worth trying log.debug?